forward_script/src/QueryBuilder.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+
+namespace Wikimedia\ForwardScript;
+
+class QueryBuilder {
+
+	public function getQuery( $info, $pageName, $id = '', $coordinates = [] ) {
+		$category = preg_replace( '/^(:?Category|Kategorie):/', '', $info->category );
+		$query = array_merge( array_filter( $coordinates ), [ 'categories' => $category ] );
+		if ( $id && empty( $info->duplicate_ids ) && empty( $info->id_not_found ) ) {
```

Comment:
An even better way is to use a value object instad of a `stdClass` from JSON and to put those tests into value object functions with proper names. I'm working on it.